library/classes.lisp

```diff
@@ -324,7 +325,20 @@ Typical `fail` continuations are:
     "Convert any Unwrappable container into an Optional, constructing Some on a successful unwrap and None on a failed unwrap."
     (unwrap-or-else Some
                     (fn () None)
-                    container)))
+                    container))
+
+
+  ;;
+  ;; Default
+  ;;
+
+  (define-class (Default :a)
+    (default (Unit -> :a)))
+
+  (declare defaulting-unwrap ((Unwrappable :container) (Default :element) =>
+                              (:container :element) -> :element))
+  (define (defaulting-unwrap container)
+    (with-default (default) container)))
```

Comment:
One last thing. `defaulting-unwrap` should call `unwrap` directly so that the default value is only constructed if needed.